File: sdk/dotnet/src/ExperimentationPlatform/Exceptions/NetworkException.cs
namespace ExperimentationPlatform.Exceptions;

/// <summary>
/// Thrown when a network-level error occurs (e.g. connection refused, DNS failure, timeout).
/// </summary>
public class NetworkException : ExperimentationException
{
    public NetworkException(string message) : base(message) { }

    public NetworkException(string message, Exception innerException)
        : base(message, innerException) { }
}

File: sdk/php/src/Errors/NetworkException.php
<?php

declare(strict_types=1);

namespace Experimently\Errors;

/**
 * Thrown when a network-level error occurs (cURL failure, timeout, etc.).
 */
class NetworkException extends ExperimentationException
{
    public function __construct(string $message = '', int $code = 0, ?\Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}
